### PR TITLE
Use Google Cloud KMS for signing

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
Minimum changes required to comply with the new [DigiCert's FIPS 140-2 Level 2 or Common Criteria EAL 4+ requirements](https://knowledge.digicert.com/general-information/new-private-key-storage-requirement-for-standard-code-signing-certificates-november-2022).

* https://icedev.pl/posts/setting-up-ev-code-signing-google-hsm-fips-140-2
* https://github.com/mattes/google-cloud-kms-csr

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -40,12 +40,25 @@ jobs:
...
-        EXE_ITERATIVE_CERTIFICATE: ${{ secrets.EXE_ITERATIVE_CERTIFICATE }}
-        EXE_ITERATIVE_CERTIFICATE_PASS: ${{ secrets.EXE_ITERATIVE_CERTIFICATE_PASS }}
-        PSModulePath: ""
-      run: python sign.py dvc-*.exe
+      run: java -jar jsign-4.2.jar dvc-*.exe
+        --storetype GOOGLECLOUD
+        --storepass $(gcloud auth print-access-token)
+        --keystore projects/dvc-artifact-signing/locations/global/keyRings/code-signing
+        --alias windows
+        --certfile codesign-chain.pem
```
Comment (0x2b3bfa0) [inline .github/workflows/build.yaml:58]: Pending: commit this file once we have it.

**Comment by skshetry** [comment]
@0x2b3bfa0, do you mind adding an `upload-artifact` to the workflow? 

I don't have a Windows machine to test with me, but we should try installing the binary on Windows and see if we get any issues.

**Review by skshetry — APPROVED** [review]

**Comment by 0x2b3bfa0** [comment]
Captura de pantalla 2024-04-03 a las 17 45 06
Captura de pantalla 2024-04-03 a las 17 46 18
Captura de pantalla 2024-04-03 a las 17 47 56

**Comment by skshetry** [comment]
Thank you @0x2b3bfa0. Let's merge it. 🙂

**Pull request merged**